python/src/gluon_ir.cc

```diff
@@ -226,17 +226,19 @@ py::object layoutToGluon(Attribute layout) {
         toStdVector(ctaLayout.getCTAOrder()));
   } else if (auto paddedShared =
                  dyn_cast<ttg::PaddedSharedEncodingAttr>(layout)) {
-    auto ctaLayout = paddedShared.getCTALayout();
+    auto *ctx = paddedShared.getContext();
     std::vector<std::pair<unsigned, unsigned>> intervalPaddingPairs;
     for (auto [interval, padding] :
          llvm::zip(paddedShared.getIntervals(), paddedShared.getPaddings())) {
       intervalPaddingPairs.push_back({interval, padding});
     }
+    auto kOffset = mlir::StringAttr::get(ctx, "offset");
+    auto kBlock = mlir::StringAttr::get(ctx, "block");
+    auto ll = paddedShared.getLinearComponent();
+    auto shape = toStdVector(ArrayRef(llvm::to_vector(ll.getOutDimSizes())));
```

Comment:
Out of curiosity, is this an idiom or does it have a redundant copy?

Reply:
You are right this is a redundant copy. I missed the `LinearLayoutParam` in the tablegen so `getLienarComponent()` returned the LL by value, hence I used `auto` instead of `const auto&`. Will be fixed in the next push.

(Also removed the same redundant copy for the DistributedLinearLayout case)